x-pack/plugins/security_solution/public/common/containers/sourcerer/index.tsx

```diff
@@ -0,0 +1,414 @@
+          payload: { defaultPatterns: defaultIndex, id },
+        });
+      }
+
+      async function fetchSource() {
+        if (!apolloClient) return;
+        setIsSourceLoading({ id, loading: true });
+        try {
+          const result = await apolloClient.query<SourceQuery.Query, SourceQuery.Variables>({
+            query: sourceQuery,
+            fetchPolicy: 'network-only',
+            variables: {
+              sourceId: 'default', // always
+              defaultIndex: selectedPatterns,
+            },
+            context: {
+              fetchOptions: {
+                signal: abortCtrl.signal,
+              },
+            },
+          });
+          if (isSubscribed) {
+            dispatch({
+              type: 'SET_SOURCE',
+              id,
+              defaultIndex: selectedPatterns,
+              payload: {
+                browserFields: getBrowserFields(
+                  selectedPatterns.join(),
+                  get('data.source.status.indexFields', result)
+                ),
+                docValueFields: getDocValueFields(
+                  selectedPatterns.join(),
+                  get('data.source.status.indexFields', result)
+                ),
+                errorMessage: null,
+                id,
+                indexPattern: getIndexFields(
+                  selectedPatterns.join(),
+                  get('data.source.status.indexFields', result)
+                ),
+                indexPatterns: selectedPatterns,
+                indicesExist: indicesExistOrDataTemporarilyUnavailable(
+                  get('data.source.status.indicesExist', result)
+                ),
+                loading: false,
+              },
+            });
+          }
+        } catch (error) {
+          if (isSubscribed) {
+            dispatch({
+              type: 'SET_SOURCE',
+              id,
+              defaultIndex: selectedPatterns,
+              payload: {
+                errorMessage: error.message,
+                id,
+                loading: false,
+              },
+            });
+          }
+        }
+      }
+
+      fetchSource();
+
+      return () => {
+        isSubscribed = false;
+        return abortCtrl.abort();
+      };
+    },
+    [
+      apolloClient,
+      getDefaultIndex,
+      setIsSourceLoading,
+      state.availableIndexPatterns,
+      state.sourceGroups,
+    ]
+  );
+
+  const initializeSourceGroup = useCallback(
+    (id: SourceGroupsType, indexToAdd?: string[] | null, onlyCheckIndexToAdd?: boolean) =>
+      enrichSource(id, indexToAdd, onlyCheckIndexToAdd),
+    [enrichSource]
+  );
+
+  const updateSourceGroupIndicies = useCallback(
+    (id: SourceGroupsType, updatedIndicies: string[]) => enrichSource(id, updatedIndicies, true),
+    [enrichSource]
+  );
+  const getManageSourceGroupById = useCallback(
+    (id: SourceGroupsType) => {
+      if (state.sourceGroups[id] != null) {
+        return state.sourceGroups[id] as ManageSource;
```

Comment:
This seems to avoid `as` on line 365 successfully.
```
const getManageSourceGroupById = useCallback(
    (id: SourceGroupsType) => {
      const sourceById = state.sourceGroups[id];
      if (sourceById != null) {
        return sourceById;
      }
      return getSourceDefaults(id, getDefaultIndex());
    },
    [getDefaultIndex, state.sourceGroups]
  );
```